File: Chapter 08/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter08/section_8_1_4_database_comparison/infrastructure/NoSqlDocument.java
package com.grokkingsoftwarearchitecture.chapter08.section_8_1_4_database_comparison.infrastructure;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class NoSqlDocument {
    public String name = "";
    public List<String> tags = new ArrayList<>();
    
    /**
     * JAVA ARCHITECTURAL NOTE: 
     * In Java, objects are strictly typed. To simulate the schema-less, 
     * flexible nature of a JSON Document DB (like MongoDB), we use a 
     * Map<String, Object> to hold arbitrary data added at runtime.
     */
    public Map<String, Object> flexibleData = new HashMap<>();
}

File: Chapter 08/C#/Section_8_1_4_DB_Comparison/Infrastructure/NoSqlDocument.cs
// Infrastructure/NoSqlDocument.cs
using System.Collections.Generic;

namespace Chapter08.DatabaseCode.Infrastructure
{
    public class NoSqlDocument
    {
        public string Name { get; set; } = string.Empty;
        public List<string> Tags { get; set; } = new List<string>();
        
        /// <summary>
        /// In C#, objects are strictly typed. To simulate the schema-less, 
        /// flexible nature of a JSON Document DB (like MongoDB), we use a 
        /// dictionary to hold arbitrary data added at runtime.
        /// </summary>
        public Dictionary<string, object> FlexibleData { get; set; } = new Dictionary<string, object>();
    }
}